plugins/providers/virtualbox/cap/configure_disks.rb

```diff
@@ -52,19 +96,24 @@ def self.configure_disks(machine, defined_disks)
         # @param [Vagrant::Machine] machine - the current machine
         # @param [Config::Disk] disk - the current disk to configure
         # @param [Array] all_disks - A list of all currently defined disks in VirtualBox
+        # @param [VagrantPlugins::ProviderVirtualBox::Model::StorageController] controller -
+        # the storage controller to use
         # @return [Hash] current_disk - Returns the current disk. Returns nil if it doesn't exist
-        def self.get_current_disk(machine, disk, all_disks)
+        def self.get_current_disk(machine, disk, all_disks, controller)
           current_disk = nil
           if disk.primary
             # Ensure we grab the proper primary disk
             # We can't rely on the order of `all_disks`, as they will not
             # always come in port order, but primary is always Port 0 Device 0.
-            vm_info = machine.provider.driver.show_vm_info
-            primary_uuid = vm_info["SATA Controller-ImageUUID-0-0"]
+            primary = controller.attachments.detect { |a| a[:port] == "0" && a[:device] == "0" }
+            if primary.nil?
+              raise Vagrant::Errors::VirtualBoxDisksPrimaryNotFound
+            end
+            primary_uuid = primary[:uuid]
 
             current_disk = all_disks.select { |d| d["UUID"] == primary_uuid }.first
           else
-            current_disk = all_disks.select { |d| d["Disk Name"] == disk.name}.first
+            current_disk = all_disks.select { |d| d["Disk Name"] == disk.name }.first
```

Comment:
Hm, maybe this could turn into a `.detect`?
```suggestion
            current_disk = all_disks.detect { |d| d["Disk Name"] == disk.name }
```